determined-ai/determined
# fix: regular integer spacing of chart ticks [WEB-1714]

## Description

Updates LineChart component and TrialDetails/TrialChart to have integer tick values on a batch or epoch-based x-axis.
uPlot accepts an array of acceptable intervals.

Unfortunately the array must have every possible tick spacing, meaning if we put [1], the only acceptable spacing is every batch. At larger scales this is too dense and no ticks are displayed. 

My idea is to set tick interval to every `1, 2, 3, 5, 10, 20, 30, 50, 100, 200, 300, 500, 1000, 2000, 3000, 5000` ... up to `1_000_000, 2_000_000` or `5_000_000`. Probably good up to 50 million batch jobs. If the 1,2,3,5 pattern is too condensed I could try 1,2,5,6 ?

## Test Plan

Verify a meaningful x-axis with integer values.
- View LineChart on `/det/design`
- View charts on a trial Overview tab
- View charts on a trial Metrics tab. Change the x-axis to Time and back to Batches.
- Zoom in on a small x-range of a chart, and it should continue to show integer values

## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
